python/prepare_release.py

```diff
@@ -0,0 +1,296 @@
+
+
+def push_branch(upcoming_version_branch):
+    print(f"Pushing changes for {upcoming_version_branch} into remote origin...### ")
+    run(f"git push --set-upstream origin {upcoming_version_branch}")
+    print(f"### OK Changes pushed for {upcoming_version_branch}### ")
+
+
+def commit_all_changes_on_branch(project_name, project_version, release_branch):
+    print(f"### Committing changes for branch {release_branch}...### ")
+    run(f' git commit -a -m "Bump  {project_name} version to  {project_version} "')
+    print(f"### OK Changes committed for {release_branch}### ")
+
+
+def update_version_with_upcoming_version_in_citus_control(citus_control_file_path, upcoming_minor_version):
+    print(f"### Updating {citus_control_file_path} file with the upcoming version {upcoming_minor_version}...### ")
+    if not replace_line_in_file(citus_control_file_path, CITUS_CONTROL_SEARCH_PATTERN,
+                                f"default_version = '{upcoming_minor_version}-1'"):
+        raise ValueError(f"{citus_control_file_path} does not have match for version")
+    print(f"### OK{citus_control_file_path} file updated with the upcoming version {upcoming_minor_version}...### ")
+
+
+def update_schema_version_with_upcoming_version_in_multi_extension_file(current_schema_version,
+                                                                        multi_extension_sql_path,
+                                                                        upcoming_minor_version, upcoming_version):
+    print(
+        f"### Updating schema version {current_schema_version} on {multi_extension_sql_path} "
+        f"file with the upcoming version {upcoming_version}...### ")
+    # TODO Append instead of replace may require
+    if not replace_line_in_file(multi_extension_sql_path,
+                                f"ALTER EXTENSION citus UPDATE TO '{current_schema_version}'",
+                                f"ALTER EXTENSION citus UPDATE TO '{upcoming_minor_version}-1';"):
+        raise ValueError(f"{multi_extension_sql_path} does not have match for version")
+    print(f"### OK Current schema version updated on {multi_extension_sql_path} to {upcoming_minor_version}### ")
+
+
+def get_current_schema_from_citus_control(citus_control_file_path):
+    print(f"### Reading current schema version from  {citus_control_file_path}...### ")
+    current_schema_version = get_current_schema_version(citus_control_file_path)
+    if len(current_schema_version) == 0:
+        raise ValueError("Version info could not be found in citus.control file")
+    print(f"### OK Current schema version is {current_schema_version}### ")
+    return current_schema_version
+
+
+def update_version_with_upcoming_version_inconfig_py(config_py_path, upcoming_minor_version):
+    print(f"### Updating {config_py_path} file with the upcoming version {upcoming_minor_version}...### ")
+    if not replace_line_in_file(config_py_path, "^MASTER_VERSION =*",
```

Comment:
Can we move `^MASTER_VERSION =*` to a constant ? (and let's be more specific, by using something else instead of `*`)

In general, can we please move such hardcoded regex strings into some constants (as we did for `MULTI_EXT_MAJOR_SEARCH_PATTERN`) ?

Reply:
minor: let's use `\.` instead of `.`

r"^MASTER_VERSION = '\d+.\d+'" -> r"^MASTER_VERSION = '\d+**\\**.\d+'"

Reply:
Fixed